layers/+lang/emacs-lisp/packages.el

```diff
@@ -61,10 +71,17 @@
           "mhh" 'elisp-slime-nav-describe-elisp-thing-at-point)))))
 
 (defun emacs-lisp/init-emacs-lisp ()
+  (add-hook 'emacs-lisp-mode-hook
+            (lambda () (auto-compile-on-save-mode 1)))
+  (add-hook 'lisp-interaction-mode-hook
+            (lambda () (auto-compile-on-save-mode 1)))
```

Comment:
Besides, I don't think that you need a `lambda` here.  `auto-compile-on-save-mode` should be a proper minor mode—if it's not, we shouldn't use the package anyway—and thus can be added directly to the hook.  This also allows to conveniently `remove-hook` it again :smile: